pilot_experiment/2_explore_corr_of_genes.ipynb

```diff
@@ -1397,188 +1405,293 @@
+    "                                              real_acc_corr_score,\n",
+    "                                              real_core_acc_corr_score,\n",
+    "                                              shuffled_all_corr_score]))\n",
+    "\n",
+    "# Set up the matplotlib figure\n",
+    "fig, axes = plt.subplots(ncols=2, nrows=2, figsize=(8,8))\n",
+    "\n",
+    "# Distribution plot for core genes\n",
+    "sns.distplot(real_core_corr_score, \n",
+    "             label='core', \n",
+    "             color='red',\n",
+    "             bins=bins_corr,\n",
+    "             kde=False,\n",
+    "             ax=axes[0,0]\n",
+    "            )\n",
+    "\n",
+    "sns.distplot(real_acc_corr_score,\n",
+    "             label='accessory',\n",
+    "             color='blue',\n",
+    "             bins=bins_corr,\n",
+    "             kde=False,\n",
+    "             ax=axes[0,1]\n",
+    "            )\n",
+    "sns.distplot(real_core_acc_corr_score, \n",
+    "             label='core-accessory', \n",
+    "             color='purple',\n",
+    "             bins=bins_corr,\n",
+    "             kde=False,\n",
+    "             ax=axes[1,0]\n",
+    "            )\n",
+    "sns.distplot(shuffled_all_corr_score,\n",
+    "             label='shuffled', \n",
+    "             color='grey',\n",
+    "             bins=bins_corr,\n",
+    "             kde=False,\n",
+    "             ax=axes[1,1]\n",
+    "            )\n",
+    "\n",
+    "plt.suptitle('Histogram of correlation scores per group')\n",
+    "axes[0,0].set_title('Core-Core')\n",
+    "axes[0,1].set_title('Accessory-Accessory')\n",
+    "axes[1,0].set_title('Core-Accessory')\n",
+    "axes[1,1].set_title('Shuffled')\n",
+    "fig.text(0.5, 0.01, 'Correlation between genes', ha='center', fontsize=12)\n",
+    "fig.text(0.01, 0.5, 'Count', ha='center', rotation=90, fontsize=12)\n",
+    "plt.tight_layout(pad=0.4, \n",
+    "                 w_pad=0.5,\n",
+    "                 h_pad=1.0,\n",
+    "                 rect=[0, 0.03, 1, 0.95])"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "**Some sanity checks**\n",
+    "**Note about visualizations:**\n",
+    "* Based on the density plot, we observed a shift in the accessory-accessory gene correlation scores. This density plot represents the probability of a random variable falling within a particular range of values (P(0 <= X <= 0.5)). This probability is given by the integral of this variable's PDF over that range -- that is it is given by the area under the density function. So there is a higher likelihood of correlation scores > 0.5 for accessory-accessory genes compared to core-core genes. But the *exact probability* of accessory-accessory genes having a correlation score > 0.5 is not known from just visually inspecting the density plot. To get the exact probability we would need to calculate the integral from 0.5 onward.\n",
```

Comment:
But you could approximate this integral in this case, right? Since your density plot is just a continuous approximation of your discrete data (gene correlations), you could just count the number of gene correlations > 0.5 and divide it by the total number of gene correlations.

Reply:
Yes that would be an approximation you could do for future.